Userland/Libraries/LibGfx/ImageFormats/TGALoader.cpp

```diff
@@ -71,41 +71,44 @@ class TGAReader {
     {
     }
 
-    ALWAYS_INLINE u8 read_u8()
+    ALWAYS_INLINE ErrorOr<u8> read_u8()
```

Comment:
Honestly, iterating over this custom reader sounds like a bad idea to me, we should put our efforts towards using more `Stream`s.
These two last commits solve bugs that are caused by this custom implementation. How many more bugs are still hidden here but already fixed in `Stream`s?